### ISSUE TITLE
Error(Custom("missing field `xmlns:soap`"), State { next_error: None, backtrace: None })

### ISSUE THREAD

**Issue by fusillicode** [RReverser/serde-xml-rs#105]
Hi, I'm facing a strange issue trying to deserialize this XML string:
```xml 
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
<soap:Body>
    <checkAnaResponse xmlns="urn:random.eu:foo:barservices:checkAna:types">
        <countryCode>DE</countryCode>
        <anaNumber>123123anaNumber>
        <requestDate>2019-06-14+02:00</requestDate>
        <valid>false</valid>
        <name/>
        <address/>
    </checkAnaResponse>
</soap:Body>
</soap:Envelope>
```
The error I'm getting is the following one:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error(Custom("missing field `soap:Body`"), State { next_error: None, backtrace: None })', src/libcore/result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
This is the actual code:
```rust
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename = "soap:Envelope")]
struct SoapEnvelope {
    #[serde(rename = "soap:Body")]
    soap_body: SoapBody,
}

#[derive(Debug, Serialize, Deserialize)]
struct SoapBody {
    #[serde(rename = "checkAnaResponse")]
    check_ana_response: CheckAnaResponse,
}

#[derive(Debug, Serialize, Deserialize)]
struct CheckAnaResponse {
    #[serde(rename = "countryCode")]
    country_code: String,
    #[serde(rename = "anaNumber")]
    ana_number: String,
    #[serde(rename = "requestDate")]
    request_date: String,
    valid: bool,
    name: String,
    address: String,
}
```
I've tried to add additional container renamings to the `SoapBody` struct but nothing changed :(

What am I missing?

**Comment by punkstarman** [comment]
This project doesn't support specifying the namespace of a tag. When deserializing, tags are stripped of any XML namespace. Thus, `soap:Envelope` becomes `Envelope`.

The following type definitions work.

```rust
#[derive(Debug, Serialize, Deserialize)]
#[serde(rename = "Envelope")]
struct SoapEnvelope {
    #[serde(rename = "Body")]
    soap_body: SoapBody,
}

#[derive(Debug, Serialize, Deserialize)]
struct SoapBody {
    #[serde(rename = "checkAnaResponse")]
    check_ana_response: CheckAnaResponse,
}

#[derive(Debug, Serialize, Deserialize)]
struct CheckAnaResponse {
    #[serde(rename = "countryCode")]
    country_code: String,
    #[serde(rename = "anaNumber")]
    ana_number: String,
    #[serde(rename = "requestDate")]
    request_date: String,
    valid: bool,
    name: String,
    address: String,
}
```


BTW:

* You're XML code contains a typo since the `anaNumber` tag is not closed
* The error reported in the title of this issue is not the one reported in the description. It is possibly an error from a previous version of your code.
* Instead of renaming each individual field of a type, you can use `rename_all` on the type itself and choose a naming policy. See https://serde.rs/container-attrs.html

**Comment by fusillicode** [comment]
I'm deeply sorry for the inaccuracies of my issue. I know that it's not an excuse but just to give a little bit of context I opened it late at night and quite a lot of errors slipped in while I tried to "sanitize" both the XML snippet and the code.

I promise I will take extra care next time.

Thanks a lot for taking the time to answer even if the issue was malformed.

Sorry once again and thank you for your work 🙇‍♂️
